### ISSUE TITLE
Exists

### ISSUE THREAD

**Issue by Gabriel-Paulucci** [SeaQL/sea-orm#408]
Add a variant of `find`, which checks if the record exists in the database but doesn't return any data, just says whether or not it exists

This improves performance in some cases where a test of whether the user has already been registered and the case will not be registered

There are better ways to do this test, which is in the insertion case of failure, check the error code, but as it is not so easy at the moment to obtain this code, it would be easier to implement this function

**Comment by RomainMazB** [comment]
This would be useful indeed, I may take a look to implement this in the next few days.

**Comment by billy1624** [comment]
How about adding a new operation `exists` under find?

```rust
let _: bool = Entity::find_by_id(1)
    .exists(db)
    .await?

let _: bool = Entity::find()
    .filter( ... )
    .exists(db)
    .await?
```

**Comment by Gabriel-Paulucci** [comment]
I think it's good like this, just one thing that maybe you should take into account, when using the filter maybe there can be more than one, but when the filter is by id, it can only exist or not

So maybe you can use `count(1)` in the filter to be able to nab the amount of existing records

**Comment by billy1624** [comment]
Btw... you could also `count` it with paginator
https://docs.rs/sea-orm/0.4.2/sea_orm/trait.PaginatorTrait.html#method.count

**Comment by Gabriel-Paulucci** [comment]
yes, I agree with what you said, but that would be one more call to the database

that's why I talked about the filter returning the number of records, but it's just my opinion

**Comment by RomainMazB** [comment]
🤔 Using the sql `EXISTS` statement, no matter if there is one or multiple results, it will just return `TRUE` or `FALSE` directly from the database.

I thought this would be already implemented into sea-query but it's not. I guess this should be the starting point.

Related to https://github.com/SeaQL/sea-query/issues/118

**Comment by Gabriel-Paulucci** [comment]
makes sense, and when I did that I didn't use this command, but looking at it from this side it only makes sense to return the boolean value

**Comment by tyt2y3** [comment]
I am still thinking whether we should add the method to `PaginatorTrait` or create a new `Trait` (which includes other methods like `max` or `min`)

**Comment by Gabriel-Paulucci** [comment]
I think it would be interesting to be separated as it doesn't have anything to do strictly with pagination, but something useful that can be used in any context.
